andela/ah-alpha
# #162948914 Users should be able to rate an article

#### What does this PR do?
This PR adds rating functionality to Authors' Haven.

#### Description of Task to be completed?
Users(readers) should be able to:

- Rate articles
- Delete their ratings on articles

#### How should this be manually tested?

* Clone ah-alpha repo ```git clone https://github.com/andela/ah-alpha.git```
* git checkout `ft-rate-articles-162948914`
* Install requirements `pip3 install -r requirements.txt`
* Run the server `python3 manage.py runserver`
* Launch postman and create a user by sending a POST request to `http://127.0.0.1:8000/api/v1/users/` with a body of:
```
{
   "user":{
        "username": "username",
        "email": "valid@email.com",
        "password": "Passw0rd"
    }
}
```
* Login by sending a POST request to `http://127.0.0.1:8000/api/v1/users/login/` with your registered email and password
```
{
   "user":{
        "email": "valid@email.com",
        "password": "Passw0rd"
    }
}
```
* Copy the returned JWT token and store it somewhere
* Use postman to create an article by sending a POST request to `http://127.0.0.1:8000/api/v1/articles`
* The body should have the format of 
```
{ 
  "article":{
       "image_path":"dan",
            "title":"rick",
            "description":"and",
            "body":"morty"
    }
}
```
* Select Bearer Token under Authorization and paste in the token gotten previously.
* The response should be similar to 
```
{
    "id": 14,
    "author": {
        "email": "ulumadan2@gmail.com",
        "username": "newuser"
    },
    "image_path": "dan",
    "slug": "rick-5",
    "title": "rick",
    "description": "and",
    "body": "morty",
    "created_at": "2019-01-23T14:51:16.512539Z",
    "updated_at": "2019-01-23T14:51:16.512733Z",
    "favourites": false
}
```
* Take note of the value of slug attribute returned in the response. We'll need it next.
* Rate the article by  sending a POST request to the url `http://127.0.0.1:8000/api/v1/rate/<slug>/` with the following in the body.
```
{
	"your_rating": 2
}
```
#### Something to NOTE: Make Sure to use a different user to rate the article since article authors can't rate their own articles.
* To Get and Delete article ratings, use GET and DELETE method respectively by passing a 
request to url `http://127.0.0.1:8000/api/v1/rate/<slug>/ `

#### Any background context you want to provide?
* This project uses the latest version of python3. Please download it from [here](https://www.python.org/downloads/)
* You may as well use pip instead of pip3 if you have either only python3 installed on your computer or you have it set as the default version of python

#### What are the relevant pivotal tracker stories?
[162948914 ](https://www.pivotaltracker.com/story/show/162948914)
